extensions/objects/templates/FilepathEditor.js

```diff
@@ -31,109 +31,118 @@ oppia.directive('filepathEditor', [
       scope: true,
       template: '<div ng-include="getTemplateUrl()"></div>',
       controller: ['$scope', function($scope) {
-        // Reset the component each time the value changes (e.g. if this is part
-        // of an editable list).
+        // Reset the component each time the value changes
+        // (e.g. if this is part of an editable list).
         $scope.$watch('$parent.value', function(newValue) {
-          $scope.localValue = {
-            label: newValue || ''
-          };
-          $scope.imageUploaderIsActive = false;
+          if (newValue) {
+            $scope.setSavedImageUrl(newValue, false);
+          }
         });
 
-        $scope.explorationId = explorationContextService.getExplorationId();
+        $scope.validate = function(state) {
+          return state.state === 'saved' &&
```

Comment:
Done. I changed to data and mode. But I prefer fields not to exist when they don't apply. So if mode is EMPTY I don't think savedUrl should be empty, it should just not exist. Thoughts?